sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/MicroBatchWrite.scala

```diff
@@ -36,8 +36,8 @@ class MicroBatchWrite(eppchId: Long, val writeSupport: StreamingWrite) extends B
     writeSupport.abort(eppchId, messages)
   }
 
-  override def createBatchWriterFactory(): DataWriterFactory = {
-    new MicroBatchWriterFactory(eppchId, writeSupport.createStreamingWriterFactory())
+  override def createBatchWriterFactory(numPartitions: Int): DataWriterFactory = {
```

Comment:
sorry to come up with this at the last minute: can we create a `PhysicalWriteInfo` interface? In case we want to add more physical information in the future.

Reply:
No problem! Since we still want to move forward with the interface-based approach, I've decided to evolve https://github.com/apache/spark/pull/25990/ to include both the `PhysicalWriteInfo` as well as `WriteInfo`